src/NLog/Layouts/JSON/JsonLayout.cs

```diff
@@ -464,6 +473,62 @@ private void AppendJsonPropertyValue(string propName, object? propertyValue, str
             }
         }
 
+        private void AppendJsonPropertyValue(string propName, object? propertyValue, StringBuilder sb, bool beginJsonMessage)
+        {
+            if (ExcludeEmptyProperties && (propertyValue is null || ReferenceEquals(propertyValue, string.Empty)))
+                return;
+
+            var initialLength = sb.Length;
+            BeginJsonProperty(sb, propName, beginJsonMessage, true);
+            if (!JsonConverter.SerializeObject(propertyValue, sb))
+            {
+                sb.Length = initialLength;
+                return;
+            }
+
+            if (ExcludeEmptyProperties && sb[sb.Length - 1] == '"' && sb[sb.Length - 2] == '"' && sb[sb.Length - 3] != '\\')
+            {
+                sb.Length = initialLength;
+            }
+        }
+        private void AppendJsonPropertyValue(string propName, object? propertyValue, string? format, IFormatProvider? formatProvider, MessageTemplates.CaptureType captureType, StringBuilder sb, bool beginJsonMessage)
+        {
+            AppendPropertyValueInternal(propName, propertyValue, format, formatProvider, captureType, sb, beginJsonMessage);
+        }
+
+        private void AppendFlattenedPropertyValue(string propName, object? propertyValue, string? format, IFormatProvider? formatProvider, MessageTemplates.CaptureType captureType, StringBuilder sb, bool beginJsonMessage)
+        {
+            if (captureType == MessageTemplates.CaptureType.Serialize || captureType == MessageTemplates.CaptureType.Stringify)
+            {
+                AppendPropertyValueInternal(propName, propertyValue, format, formatProvider, captureType, sb, beginJsonMessage);
+            }
+            else
+            {
+                var serializer = ResolveService<IJsonConverter>() as Targets.DefaultJsonSerializer;
```

Comment:
Was hoping that any handling of `DottedRecursion` would be handled by the `IJsonConverter` would having to cast into `DefaultJsonSerializer`.